corehq/motech/repeaters/models.py

```diff
@@ -1327,3 +1381,11 @@ def domain_can_forward(domain):
         domain_has_privilege(domain, ZAPIER_INTEGRATION)
         or domain_has_privilege(domain, DATA_FORWARDING)
     )
+
+
+@quickcache(['domain'], timeout=60)
```

Comment:
Can we remove this and only add it later if we find out that it's needed?

Reply:
> or is the idea that with a short timeout, it isn't a big deal if we operate with stale data for less than a minute?

Yes, that.

> Can we remove this and only add it later if we find out that it's needed?

Could you say more about this suggestion? We call `domain_can_forward_now()` many times a second, and it calls `toggles.PAUSE_DATA_FORWARDING.enabled()`. The consequence of caching it is that it takes under a minute to pause or unpause data forwarding for a domain, but it saves what looks like an expensive function call. I'm not sure what you mean by "if ... it's needed". In terms of performance, it seems better to cache it than not to cache it. What am I not considering? What is the downside of caching it?

Reply:
> I'm not sure what you mean by "if ... it's needed".

I meant after we have determined that performance is bad without caching, which would indicate that we need it.

> We call `domain_can_forward_now()` many times a second, and it calls `toggles.PAUSE_DATA_FORWARDING.enabled()`. The consequence of caching it is that it takes under a minute to pause or unpause data forwarding for a domain, but it saves what looks like an expensive function call.

This made me think you have measured performance and determined that we need caching. If that's true, then you can ignore my request.

> In terms of performance, it seems better to cache it than not to cache it.

However, this part made me wonder if we just added caching because "it seems better" (without measuring performance first).

> What is the downside of caching it?

Complexity. Cached values can be hard to reason about, especially in asynchronous/concurrent systems.

Reply:
4955ef4a9628c5238956ea07dc6e592cc73b65a6